lib/plugin.js

```diff
@@ -13,332 +13,338 @@ const debug = require('debug')('hh-plugin')
 exports.register = function (server, opts, next) {
     server.expose('version', require('../package.json').version);
 
+    const adapter = opts.adapter
+    adapterUtils.checkValidAdapter(adapter)
+    server.expose('adapter', adapter)
 
-    var adapter = establishAdapter();
-
-    function establishAdapter() {
-        if (!opts.adapter) {
-            // if no adapter has been passed in, fallback to a default Mongodb
-            // the Mongodb adapter will look for a running database on localhost and DOCKER_HOST port 27017
-            debug(`adapter not specified, fallback to default 'mongodb' adapter`)
-            return adapterUtils.getStandardAdapter('mongodb')()
-        } else {
-            adapterUtils.checkValidAdapter(opts.adapter)
-            return opts.adapter
-        }
-    }
-
-    adapter.connect(() => {
-        server.expose('adapter', adapter);
-        next()
-    });
-
-    const schemas = {};
-
-    const appendLinkedResources = includes(server, adapter, schemas).appendLinkedResources;
-
-    const onRouteRegister = function (schema, createRelationshipsRoutesRead, createRelationshipsRoutesWrite) {
-        routeUtils.createOptionsRoute(server, schema)
-        if (createRelationshipsRoutesRead) {
-            routeUtils.createRelationshipsRoutesRead(server, schema)
-        }
-        if (createRelationshipsRoutesWrite) {
-            routeUtils.createRelationshipsRoutesWrite(server, schema)
-        }
-        adapter.processSchema(schema)
-        schemas[schema.type] = schema
-    }
-
-    const get = function (schema) {
-        onRouteRegister(schema);
-        return _.merge(routes.get(schema), {
-            handler: (req, reply) => {
-                routeUtils.parseComparators(req)
-
-                const query = req.query
-                const filter = query.filter
-                const limit = (query.page && query.page.limit) || 1000
-                const skip = (query.page && query.page.offset) || 0
-                const sort = query.sort
-
-                var fields = extractFields(query, schema);
-
-                reply(adapter.find(schema.type, filter, skip, limit, sort, fields).then((data)=> {
-                    return {data: data}
-                }))
+    const adapterSSE = opts.adapterSSE
+    adapterUtils.checkValidAdapterSSE(adapterSSE)
+    server.expose('adapterSSE', adapterSSE)
+
+    Promise.all([
+            adapter.connect(),
+            adapterSSE.connect()
+        ])
+        .then(()=> {
+
+            const schemas = {};
+
+            const appendLinkedResources = includes(server, adapter, schemas).appendLinkedResources;
+
+            const onRouteRegister = function (schema, createRelationshipsRoutesRead, createRelationshipsRoutesWrite) {
```

Comment:
I might change this, I dislike arguments, we should just use objects for anything more than one arg. (I know it's not your code)